src/sweetalert2.js

```diff
@@ -1025,17 +1027,36 @@ sweetAlert.argsToParams = (args) => {
       break
 
     case 'object':
-      showWarningsForParams(args[0])
       Object.assign(params, args[0])
       break
 
     default:
-      error('Unexpected type of argument! Expected "string" or "object", got ' + typeof args[0])
-      return false
+      throw new TypeError('Unexpected type of argument! Expected "string" or "object", got ' + typeof args[0])
   }
   return params
 }
 
+sweetAlert.plugin = function (plugin) {
```

Comment:
For people who will read the source code, it would be super-handy to have a JSDoc with the description and the code example(s) of `swal.plugin()` usage. Basically, copy-paste from the perfect PR description.